### PR TITLE
FEATURE: license checker

### PR DESCRIPTION
**Pull request by Segelzwerg**

### CODE REVIEW

File: liccheck/action.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    - name: Freeze Dependencies
+      shell: bash
+      run: poetry export --without-hashes --format=requirements.txt > requirements.txt
+
+    - name: Check Licenses
+      id: license-check
+      run: |
+        echo "AUTHORIZED_PACKAGES=$(poetry run liccheck -s ${{ inputs.strategy }} -r requirements.txt | grep -A 1 "check authorized packages" | grep -vE "packages..." | grep -oP "^\S+")" >> "$GITHUB_OUTPUT"
+        echo "UNKNOW_PACKAGES=$(poetry run liccheck -s .${{ inputs.strategy }} -r requirements.txt | grep -A 1 "check unknown package"s | grep -vE "packages..." | grep -oP "^\S+")" >> "$GITHUB_OUTPUT"
```
Comment (iTitus) [inline liccheck/action.yml]: Typo here. Are all usages of this variable correct?

**Review by iTitus — COMMENTED** [review]
What exactly does this check for?

**Comment by Segelzwerg** [comment]
> What exactly does this check for?

It will check if the licenses of the dependencies are compliant with policies pre defined in the repository.
E.g., you can allow MIT, BSD but do not allow GPL. https://pypi.org/project/liccheck/

File: liccheck/action.yml (hunk 1 above)
Comment (Segelzwerg) [reply liccheck/action.yml]: > Typo here. Are all usages of this variable correct?

found one `s` after ". Anything else?

File: liccheck/action.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    - name: License Unauthorized
+      uses: actions/github-script@v7
+      if: ${{ steps.license-check.outputs.UNAUTHORIZED_PACKAGES != '' }}
+      with:
+        script: |
+          if (${{ steps.license-check.outputs.UNAUTHORIZED_PACKAGES }} > 0) {
+              core.setFailed('${{ steps.license-check.outputs.UNAUTHORIZED_PACKAGES }} unauthorized packages found.');
+          }
+
```
Comment (iTitus) [inline liccheck/action.yml]: Remove this new line please

**Review by iTitus — CHANGES_REQUESTED** [review]

**Review by iTitus — APPROVED** [review]

**Pull request merged**